src/pymor/algorithms/simplify.py

```diff
@@ -0,0 +1,221 @@
+# This file is part of the pyMOR project (http://www.pymor.org).
+# Copyright 2013-2019 pyMOR developers and contributors. All rights reserved.
+# License: BSD 2-Clause License (http://opensource.org/licenses/BSD-2-Clause)
+
+import numpy as np
+from numbers import Number
+
+from pymor.algorithms.lincomb import assemble_lincomb
+from pymor.algorithms.rules import RuleTable, match_class, match_generic, match_class_all, match_class_any
+from pymor.core.exceptions import RuleNotMatchingError, NoMatchingRuleError
+from pymor.models.basic import StationaryModel
+from pymor.operators.constructions import IdentityOperator, Concatenation, ConstantOperator, LincombOperator
+from pymor.operators.interfaces import OperatorInterface
+from pymor.operators.numpy import NumpyMatrixOperator
+from pymor.vectorarrays.numpy import NumpyVectorArray
+
+
+def simplify(obj, assemble=True):
+    """Simplification of various constructs, e.g. concatenation of |Operator|s.
+
+    How the simplification is realized will depend on the given object,
+    the exact algorithm is specified in :class:`SimplifyRules`.
```

Comment:
I think before considering adding this to pyMOR we should have some concept as to what `simplify` is supposed to do. E.g. what should happen with a `Concatenation` of linear `LincombOperator`. Maybe we need an option as to what is supposed to happen. What should happen with a `Concatenation` of a linear, non-parametric `Operator` and a `VectorArrayOperator`?